src/ipahealthcheck/core/config.py

```diff
@@ -63,7 +63,7 @@ def merge(self, d):
         """
         Merge variables from dict ``d`` into the configuration
 
-        The last one wins.
+        The first one wins.
```

Comment:
Isn't it the last one? If both dicts have the same key, the value will be updated to the value for the key in the passed dict.

Reply:
This is correct. The reason is because of the default cli options. I wasn't able to find a reliable way to tell whether the option had been passed in vs the argparse default value. Since pki also uses the framework I can't drop all default values because they may use them as well.

The tests verify this in an indirect way. e.g. setting output_type=human vs the default value of json.

Reply:
But I don't understand the comment. Isn't the last one winning?

Reply:
It's because of the `object.__setattr__(self, key, value)` in __setitem__. It basically makes it immutable.

Reply:
Thanks, missed that.